### ISSUE TITLE
Media upload return unknown error and second have no response

### ISSUE THREAD

**Issue by laputaworkuk** [umbraco/Umbraco-CMS#14310]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.4

### Bug summary

In appsetting, I have add 
`"Runtime": {
        "MaxQueryStringLength": 90,
        "MaxRequestLength": 1000
      }`
which restrict upload over 1mb
The first time, when I upload 3mb file, it return
Error: An unknown error occurred
Max file size is "1000KB"
as expected, but the second time, it have no response, when I inspected, the console have provide error like below :

```
`data":{"ExceptionMessage":"Object reference not set to an instance of an object.","ExceptionType":"System.NullReferenceException, 
System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","StackTrace":"   
at Umbraco.Cms.Web.BackOffice.Controllers.MediaController.PostAddFile(String path, String currentFolder, String contentTypeAlias, List`1 file)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\r\n   
at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)"},"status":500,"config":{"method":"POST","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"/umbraco/backoffice/umbracoapi/media/PostAddFile","fields":{"currentFolder":"11025","contentTypeAlias":"umbracoAutoSelect","propertyAlias":"umbracoFile"},"file":{"$error":"maxSize","$errorMessages":{"maxSize":true},"$errorParam":"1000KB","key":"04045f2b-fc30-4309-9b51-b9b1c6324901","messages":[]},"_isDigested":true,"_chunkSize":null,"headers":{"Accept":"application/json, text/plain, */*","X-Requested-With":"XMLHttpRequest","X-UMB-CULTURE":"en-GB","X-UMB-SEGMENT":null,"X-UMB-XSRF-TOKEN":"CfDJ8KHpP_QJKXFMo0vhgKjfj8vdhZvh9p3LDXtPeU4iBVidfCeAWGx_iSqMuJ8Zjs-gx8Dn1ZS3XHzO0iIkq5f7mqWYDL75-TggfqlOkK4cvK0kKwDVBKohy1_kOuvy4xlGX1aLbxrbfCotXaY4iDQaYM5M3mmTV0cztcWx57TRIHmE5-Q2usApF4Vudiyd5Zbh9g"},"_deferred":{"promise":{}}},"statusText":"","xhrStatus":"complete"}
(anonymous) @ angular.js:15697`
```

### Steps to reproduce

In the appsetting
the configuration is 
```
"CMS": {
      "RuntimeMinification": {
        "UseInMemoryCache": true,
        "CacheBuster": "Timestamp"
      },
      "Runtime": {
        "MaxQueryStringLength": 90,
        "MaxRequestLength": 1000
      }
    }
  }
```
Upload the first big file can return error, but the second have no response

### Expected result / actual result

It should always return error if file is oversize

**Comment by laputaworkuk** [comment]
Hi @elit0451 
Thank you for your help, if you have any update, please let me know
Thank you

**Comment by Sander-Los** [comment]
We seem to be having the same issue on Umbraco 10.4.0 (and not seeing a fix in the releases afterwards)

**Comment by laputaworkuk** [comment]
This will impact on limited upload image size feature, or if someone upload too large image
it will restart the service

**Comment by elit0451** [comment]
It seems like on 10.4, we get the error below each time we upload a bigger image. 

And the reason why you @Sander-Los, don't see a fix in the releases afterwards is that as far as I know, this issue is not a regression one and therefore, it is marked as up for grabs. There doesn't seem to be a fix for it yet, as no one has picked it up yet 🙂 
![image (2)](https://github.com/umbraco/Umbraco-CMS/assets/21998037/e820df5e-6777-4ad7-a209-d7f8fa85230f)

**Comment by elit0451** [comment]
@laputaworkuk We are looking into this issue

**Comment by iOvergaard** [comment]
Fixed for V10 with #14578 and will be available in the next release.
Fixed for V12 with #14579 and will be available in 12.1.0.

Cherry-picked for V11 with 24844c6ce25c66c113e7b6ed2f9f900c624a0cdb and will be available in the (as of yet unknown) next release.
